perfkitbenchmarker/timed_interval.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A measured interval that generates timing samples."""
+
+from contextlib import contextmanager
+import time
+
+from perfkitbenchmarker import sample
+
+
+class TimedInterval(object):
+  """A measured interval that generates timing samples.
+
+  Attributes:
+    name: A string used to prefix generated samples' names.
+    start_time: A float containing the time in seconds since the epoch, recorded
+      at the beginning of a call to Measure, or None if Measure was not called.
+    stop_time: A float containing the time in seconds since the epoch, recorded
+      at the end of a call to Measure, or None if Measure was not called.
+  """
+
+  def __init__(self, name):
+    """Create a named interval.
+
+    Args:
+      name: A string used to prefix the generated samples' names.
+    """
+    self._name = name
+    self._start_time = None
+    self._stop_time = None
+
+  @property
+  def name(self):
+    return self._name
+
+  @property
+  def start_time(self):
+    return self._start_time
+
+  @property
+  def stop_time(self):
+    return self._stop_time
```

Comment:
Nit: this is a bit verbose. Generally we would just make these public attributes rather than properties.